frontend/src/store/modules/shoots/getters.js

```diff
@@ -0,0 +1,247 @@
+      return metadata.creationTimestamp
+    case 'createdBy':
+      return getCreatedBy(metadata)
+    case 'project':
+      return getProjectName(metadata)
+    case 'k8sVersion':
+      return get(spec, 'kubernetes.version')
+    case 'infrastructure':
+      return `${get(spec, 'provider.type')} ${get(spec, 'region')}`
+    case 'seed':
+      return get(item, 'spec.seedName')
+    case 'ticketLabels': {
+      const labels = rootGetters.ticketsLabels(metadata)
+      return join(map(labels, 'name'), ' ')
+    }
+    default: {
+      if (startsWith(column, 'Z_')) {
+        const path = get(rootGetters.shootCustomFields, [column, 'path'])
+        return get(item, path)
+      }
+      return metadata[column]
+    }
+  }
+}
+
+const getSortVal = (rootGetters, item, sortBy) => {
+  const value = getRawVal(rootGetters, item, sortBy)
+  const status = item.status
+  switch (sortBy) {
+    case 'purpose':
+      switch (value) {
+        case 'infrastructure':
+          return 0
+        case 'production':
+          return 1
+        case 'development':
+          return 2
+        case 'evaluation':
+          return 3
+        default:
+          return 4
+      }
+    case 'lastOperation': {
+      const operation = value || {}
+      const inProgress = operation.progress !== 100 && operation.state !== 'Failed' && !!operation.progress
+      const lastErrors = get(item, 'status.lastErrors', [])
+      const isError = operation.state === 'Failed' || lastErrors.length
+      const allErrorCodes = errorCodesFromArray(lastErrors)
+      const userError = isUserError(allErrorCodes)
+      const ignoredFromReconciliation = isReconciliationDeactivated(get(item, 'metadata', {}))
+
+      if (ignoredFromReconciliation) {
+        if (isError) {
+          return 400
+        } else {
+          return 450
+        }
+      } else if (userError && !inProgress) {
+        return 200
+      } else if (userError && inProgress) {
+        const progress = padStart(operation.progress, 2, '0')
+        return `3${progress}`
+      } else if (isError && !inProgress) {
+        return 0
+      } else if (isError && inProgress) {
+        const progress = padStart(operation.progress, 2, '0')
+        return `1${progress}`
+      } else if (inProgress) {
+        const progress = padStart(operation.progress, 2, '0')
+        return `6${progress}`
+      } else if (isShootStatusHibernated(status)) {
+        return 500
+      }
+      return 700
+    }
+    case 'readiness': {
+      const errorConditions = filter(get(status, 'conditions'), condition => get(condition, 'status') !== 'True')
+      const lastErrorTransitionTime = head(orderBy(map(errorConditions, 'lastTransitionTime')))
+      return lastErrorTransitionTime
+    }
+    case 'ticket': {
+      const { namespace, name } = item.metadata
+      return rootGetters.latestUpdatedTicketByNameAndNamespace({ namespace, name })
+    }
+    default:
+      return toLower(value)
+  }
+}
+
+export default {
```

Comment:
I would use both [named export](https://developer.mozilla.org/en-us/docs/Web/JavaScript/Reference/Statements/export#using_named_exports) and [default export](https://developer.mozilla.org/en-us/docs/Web/JavaScript/Reference/Statements/export#using_the_default_export).

```js
export function filteredItems (state) {
    return state.filteredShoots
}
export function itemByNameAndNamespace (state) {
   return findItem(state)
}
...
export default {
  filteredItems,
  itemByNameAndNamespace,
  ... 
}

or

function filteredItems (state) {
    return state.filteredShoots
}
function itemByNameAndNamespace (state) {
   return findItem(state)
}
export {
  getRawVal,
  getSortVal,
  filteredItems,
  itemByNameAndNamespace,
  ... 
}
export default {
  filteredItems,
  itemByNameAndNamespace,
  ... 
}
```

In the named exports we could export all functions also `getRawVal` and `getSortVal` e.g. makes it easier to test them. But in the default export only those which follow the getter signature and could / should be added to the store.